### PR TITLE
jbrowse 2.12.2

### PR DESCRIPTION
**Pull request by khipp**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

**Comment by reitermarkus** [comment]
The default `git` strategy seems to work fine, no?

**Comment by khipp** [comment]
Maybe I missed something, but when checking for outdated casks, I got the result `1.16.11-release`.

**Comment by khipp** [comment]
When I remove

```ruby
  livecheck do
    url :url
    strategy :github_latest
  end
```

I get the following output:

```sh
❯ brew livecheck --debug jbrowse                                                                                                                                                                        (torch) 
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::NullLoader): loading jbrowse
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromNameLoader): loading jbrowse

Cask:             jbrowse
Livecheckable?:   No

URL:              https://github.com/GMOD/jbrowse-components/releases/download/v2.12.2/jbrowse-desktop-v2.12.2-mac.dmg
URL (processed):  https://github.com/GMOD/jbrowse-components.git
Strategy:         Git

Matched Versions:
0.0.1-testrelease.1, 0.0.1-testrelease.2, 1.0.0, 3@1.0.0, 1.0.1, 1.0.2, 1.0.3, 1.0.4, 1.1.0, 1.2.0, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.3.5, 1.4.0, 1.4.1, 1.4.2, 1.4.3, 1.4.4, 1.5.0, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.5.5, 1.5.6, 1.5.7, 1.5.8, 1.5.9, 1.6.0, 1.6.1, 1.6.2, 1.6.3, 1.6.4, 1.6.5, 1.6.6, 1.6.7, 1.6.8, 1.6.9, 1.7.0, 1.7.1, 1.7.10, 1.7.11, 1.7.2, 1.7.3, 1.7.4, 1.7.5, 1.7.6, 1.7.7, 1.7.8, 1.7.9, 2.0.0, 2.0.1, 2.1.0, 2.1.1, 2.1.2, 2.1.3, 2.1.4, 2.1.5, 2.1.6, 2.1.7, 2.10.0, 2.10.1, 2.10.2, 2.10.3, 2.11.0, 2.11.1, 2.11.2, 2.12.0, 2.12.1, 2.12.2, 2.2.0, 2.2.1, 2.2.2, 2.3.0, 2.3.1, 2.3.2, 2.3.3, 2.3.4, 2.4.0, 2.4.1, 2.4.2, 2.5.0, 2.6.1, 2.6.2, 2.6.3, 2.7.0, 2.7.1, 2.7.2, 2.8.0, 2.9.0

jbrowse: 2.12.2 ==> 3@1.0.0
```

I would require some assistance if there is a better way without requiring the extra `livecheck`.

File: Casks/j/jbrowse.rb [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
+  end
+
+  depends_on macos: ">= :catalina"
+
+  app "JBrowse #{version.major}.app"
+
+  zap trash: [
+    "~/Library/Application Support/@jbrowse",
+    "~/Library/Application Support/com.apple.sharedfilelist/com.apple.LSSharedFileList.ApplicationRecentDocuments/org.jbrowse2.app.sfl*",
```
Comment (reitermarkus) [inline Casks/j/jbrowse.rb:23]:
```suggestion
    "~/Library/Application Support/com.apple.sharedfilelist/com.apple.LSSharedFileList.ApplicationRecentDocuments/org.jbrowse#{version.major}.app.sfl*",
```

File: Casks/j/jbrowse.rb [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
+    "~/Library/Preferences/org.jbrowse2.app.plist",
```
Comment (reitermarkus) [inline Casks/j/jbrowse.rb:24]:
```suggestion
    "~/Library/Preferences/org.jbrowse#{version.major}.app.plist",
```

**Review by razvanazamfirei — APPROVED** [review]

**Pull request merged**